File: superstitioMain/java/superstitio/cards/general/TempCard/VulnerableTogether.java
package superstitio.cards.general.TempCard;

import basemod.cardmods.ExhaustMod;
import basemod.cardmods.RetainMod;
import basemod.helpers.CardModifierManager;
import com.megacrit.cardcrawl.characters.AbstractPlayer;
import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
import com.megacrit.cardcrawl.monsters.AbstractMonster;
import com.megacrit.cardcrawl.powers.VulnerablePower;
import superstitio.DataManager;
import superstitio.cards.general.AbstractTempCard;
import superstitioapi.utils.ActionUtility;

import java.util.Arrays;


public class VulnerableTogether extends AbstractTempCard {
    public static final String ID = DataManager.MakeTextID(VulnerableTogether.class);

    public static final CardType CARD_TYPE = CardType.SKILL;

    public static final CardRarity CARD_RARITY = CardRarity.SPECIAL;

    public static final CardTarget CARD_TARGET = CardTarget.ALL;
    private static final int COST = 0;
    private static final int MAGIC = 1;
    private static final int UPGRADE_MAGIC = 1;

    public VulnerableTogether() {
        super(ID, CARD_TYPE, COST, CARD_RARITY, CARD_TARGET);
        this.setupMagicNumber(MAGIC, UPGRADE_MAGIC);
        CardModifierManager.addModifier(this, new ExhaustMod());
        CardModifierManager.addModifier(this, new RetainMod());
    }

    @Override
    public void use(AbstractPlayer player, AbstractMonster monster) {
        addToBot_applyPower(new VulnerablePower(AbstractDungeon.player, 1, false));
        Arrays.stream(ActionUtility.getAllAliveMonsters()).forEach(creature -> {
            addToBot_applyPower(new VulnerablePower(creature, 1, false));
        });
    }

    @Override
    public void upgradeAuto() {
    }
}

File: superstitioMain/java/superstitio/cards/general/TempCard/VulnerableTogether.kt
package superstitio.cards.general.TempCard

import com.megacrit.cardcrawl.characters.AbstractPlayer
import com.megacrit.cardcrawl.dungeons.AbstractDungeon
import com.megacrit.cardcrawl.monsters.AbstractMonster
import com.megacrit.cardcrawl.powers.VulnerablePower
import superstitio.DataManager
import superstitio.SPTT_Color
import superstitio.cards.SetCardColor
import superstitio.cards.general.AbstractTempCard
import superstitioapi.cards.addExhaustMod
import superstitioapi.cards.addRetainMod
import superstitioapi.utils.CreatureUtility
import java.util.*

@SetCardColor(SPTT_Color.TempColor)
class VulnerableTogether : AbstractTempCard(ID, CARD_TYPE, COST, CARD_RARITY, CARD_TARGET)
{
    init
    {
        this.setupMagicNumber(MAGIC, UPGRADE_MAGIC)
        this.addExhaustMod()
        this.addRetainMod()
    }

    override fun use(player: AbstractPlayer?, monster: AbstractMonster?)
    {
        addToBot_applyPower(VulnerablePower(AbstractDungeon.player, this.magicNumber, false))
        Arrays.stream(CreatureUtility.getAllAliveMonsters()).forEach { creature: AbstractMonster? ->
            addToBot_applyPower(VulnerablePower(creature, this.magicNumber, false))
        }
    }

    override fun upgradeAuto()
    {
    }

    companion object
    {
        val ID: String = DataManager.MakeTextID(VulnerableTogether::class.java)

        val CARD_TYPE: CardType = CardType.SKILL

        val CARD_RARITY: CardRarity = CardRarity.SPECIAL

        val CARD_TARGET: CardTarget = CardTarget.ALL
        private const val COST = 0
        private const val MAGIC = 1
        private const val UPGRADE_MAGIC = 1
    }
}
